tests/phpunit/tests/test-translated-post.php

```diff
@@ -224,4 +224,36 @@ public function test_dont_save_translations_with_incorrect_language() {
 
 		$this->dont_save_translations_with_incorrect_language( $model->post );
 	}
+
+	/**
+	 * @ticket #1698 see {https://github.com/polylang/polylang-pro/issues/1698}.
+	 * @covers PLL_Translated_Post::get_db_infos()
+	 */
+	public function test_get_db_infos() {
+		$options = array_merge( PLL_Install::get_default_options(), array( 'default_lang' => 'en' ) );
+		$model = new PLL_Model( $options );
+
+		$ref = new ReflectionMethod( $model->post, 'get_db_infos' );
+		$ref->setAccessible( true );
+		$db_infos = $ref->invoke( $model->post );
+
+		$this->assertSame( $GLOBALS['wpdb']->posts, $db_infos['table'], 'get_db_infos() does not return the right table name.' );
+		$this->assertSame( $GLOBALS['wpdb']->posts, $db_infos['default_alias'], 'get_db_infos() does not return the right field alias.' );
+
+		if ( ! is_multisite() ) {
+			return;
+		}
+
+		$site_id = self::factory()->blog->create();
+
+		switch_to_blog( $site_id );
+		$multi_db_infos = $ref->invoke( $model->post );
+
+		$this->assertSame( $GLOBALS['wpdb']->posts, $multi_db_infos['table'], 'get_db_infos() does not return the right table name.' );
+		$this->assertSame( $GLOBALS['wpdb']->posts, $multi_db_infos['default_alias'], 'get_db_infos() does not return the right field alias.' );
+		$this->assertNotSame( $db_infos['table'], $multi_db_infos['table'], 'The table name should be different between blogs.' );
+		$this->assertNotSame( $db_infos['default_alias'], $multi_db_infos['default_alias'], 'The field alias should be different between blogs.' );
+
+		restore_current_blog();
```

Comment:
Should `restore_current_blog()` be in `tear_down()`?

Reply:
Good idea. Done in [5bb5817](https://github.com/polylang/polylang/pull/1301/commits/5bb5817f98a3db739fce9ef8705e0e926e339cdd).